roachtest: hibernate failed

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7879902?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7879902?buildTab=artifacts#/hibernate) on master @ [ec095bc2fdbe4e518b076db20e4920fab67222bf](https://github.com/cockroachdb/cockroach/commits/ec095bc2fdbe4e518b076db20e4920fab67222bf):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1017-gec095bc2fd
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22238

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7895024?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7895024?buildTab=artifacts#/hibernate) on master @ [24854994805cede37e6845ee2a94e10272b5506b](https://github.com/cockroachdb/cockroach/commits/24854994805cede37e6845ee2a94e10272b5506b):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1066-g2485499480
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7909420?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7909420?buildTab=artifacts#/hibernate) on master @ [f2b00e8039af6ea8887ec124dad8daf19da6fbf1](https://github.com/cockroachdb/cockroach/commits/f2b00e8039af6ea8887ec124dad8daf19da6fbf1):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1113-gf2b00e8039
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7916359?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7916359?buildTab=artifacts#/hibernate) on master @ [c050c9b4b57ecf2ceb5d449c31c617fe12c920e0](https://github.com/cockroachdb/cockroach/commits/c050c9b4b57ecf2ceb5d449c31c617fe12c920e0):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1120-gc050c9b4b5
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7923308?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7923308?buildTab=artifacts#/hibernate) on master @ [942a4d468e9c8ad0ef45a7be33f0a326dfb19fef](https://github.com/cockroachdb/cockroach/commits/942a4d468e9c8ad0ef45a7be33f0a326dfb19fef):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1126-g942a4d468e
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7943969?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7943969?buildTab=artifacts#/hibernate) on master @ [a80652b2e4691ea76ea49e797b1b9e0998e1d61f](https://github.com/cockroachdb/cockroach/commits/a80652b2e4691ea76ea49e797b1b9e0998e1d61f):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1152-ga80652b2e4
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7952891?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7952891?buildTab=artifacts#/hibernate) on master @ [052acc88ad9d7296ce6b8b441627fb469cc74d95](https://github.com/cockroachdb/cockroach/commits/052acc88ad9d7296ce6b8b441627fb469cc74d95):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1246-g052acc88ad
Tests run against hibernate 5.6.9
8406 Total Tests Run
8387 tests passed
19 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.serialization.SessionFactorySerializationTest.testUnNamedSessionFactorySerialization - unknown (unexpected)
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
--- FAIL: org.hibernate.serialization.SessionFactorySerializationTest.testNamedSessionFactorySerialization - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-sessions branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7969261?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7969261?buildTab=artifacts#/hibernate) on master @ [d6f98e90684894fd36f53596e6aac355676d232e](https://github.com/cockroachdb/cockroach/commits/d6f98e90684894fd36f53596e6aac355676d232e):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1303-gd6f98e9068
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-sessions branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7985118?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7985118?buildTab=artifacts#/hibernate) on master @ [dcdf599e9eaee8010137167f419509ccb7627406](https://github.com/cockroachdb/cockroach/commits/dcdf599e9eaee8010137167f419509ccb7627406):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1364-gdcdf599e9e
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-sessions branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8000023?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8000023?buildTab=artifacts#/hibernate) on master @ [7c8e7012b73a598f3aa1da9bbf422e833706f5d3](https://github.com/cockroachdb/cockroach/commits/7c8e7012b73a598f3aa1da9bbf422e833706f5d3):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1403-g7c8e7012b7
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-sessions branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8006740?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8006740?buildTab=artifacts#/hibernate) on master @ [93ed65565357538c9048ff45c878a493f2ed9b45](https://github.com/cockroachdb/cockroach/commits/93ed65565357538c9048ff45c878a493f2ed9b45):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1407-g93ed655653
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-sessions branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.hibernate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8014298?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8014298?buildTab=artifacts#/hibernate) on master @ [94e03c016955dfd64d4e15358ea92226a8f362aa](https://github.com/cockroachdb/cockroach/commits/94e03c016955dfd64d4e15358ea92226a8f362aa):


```
test artifacts and logs in: /artifacts/hibernate/run_1
(test_impl.go:297).Fatalf: 
Tests run on Cockroach v23.1.0-alpha.00000000-1415-g94e03c0169
Tests run against hibernate 5.6.9
8406 Total Tests Run
8389 tests passed
17 tests failed
1928 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: org.hibernate.jpa.test.criteria.selectcase.SelectCaseLiteralHandlingBindTest.whereCaseExpression - unknown (unexpected)
For a full summary look at the hibernate artifacts
An updated blocklist (hibernateBlockList23_1) is available in the artifacts' hibernate log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #93493 roachtest: hibernate failed [C-test-failure O-roachtest O-robot T-sql-sessions branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*hibernate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/sem/tree/type_check.go
+++ b/pkg/sql/sem/tree/type_check.go
@@ -2407,10 +2407,11 @@ func typeCheckSameTypedExprs(
 
 	switch {
 	case resolvableIdxs.Empty() && constIdxs.Empty():
-		if err := typeCheckSameTypedPlaceholders(s, desired); err != nil {
+		typ, err := typeCheckSameTypedPlaceholders(s, desired)
+		if err != nil {
 			return nil, nil, err
 		}
-		return typedExprs, desired, nil
+		return typedExprs, typ, nil
 	case resolvableIdxs.Empty():
 		return typeCheckConstsAndPlaceholdersWithDesired(s, desired)
 	default:
@@ -2464,7 +2465,7 @@ func typeCheckSameTypedExprs(
 			}
 		}
 		if !placeholderIdxs.Empty() {
-			if err := typeCheckSameTypedPlaceholders(s, firstValidType); err != nil {
+			if _, err := typeCheckSameTypedPlaceholders(s, firstValidType); err != nil {
 				return nil, nil, err
 			}
 		}
@@ -2473,15 +2474,16 @@ func typeCheckSameTypedExprs(
 }
 
 // Used to set placeholders to the desired typ.
-func typeCheckSameTypedPlaceholders(s typeCheckExprsState, typ *types.T) error {
+func typeCheckSameTypedPlaceholders(s typeCheckExprsState, typ *types.T) (*types.T, error) {
 	for i, ok := s.placeholderIdxs.Next(0); ok; i, ok = s.placeholderIdxs.Next(i + 1) {
 		typedExpr, err := typeCheckAndRequire(s.ctx, s.semaCtx, s.exprs[i], typ, "placeholder")
 		if err != nil {
-			return err
+			return nil, err
 		}
 		s.typedExprs[i] = typedExpr
+		typ = typedExpr.ResolvedType()
 	}
-	return nil
+	return typ, nil
 }
 
 // Used to type check constants to the same type. An optional typ can be
@@ -2575,7 +2577,8 @@ func typeCheckConstsAndPlaceholdersWithDesired(
 		return nil, nil, err
 	}
 	if !s.placeholderIdxs.Empty() {
-		if err := typeCheckSameTypedPlaceholders(s, typ); err != nil {
+		typ, err = typeCheckSameTypedPlaceholders(s, typ)
+		if err != nil {
 			return nil, nil, err
 		}
 	}

--- a/pkg/sql/sem/tree/type_check_test.go
+++ b/pkg/sql/sem/tree/type_check_test.go
@@ -465,3 +465,55 @@ func TestTypeCheckCollatedString(t *testing.T) {
 	require.Equal(t, rightTyp.Family(), types.CollatedStringFamily)
 	require.Equal(t, rightTyp.Locale(), "en-US-u-ks-level2")
 }
+
+func TestTypeCheckCaseExprWithPlaceholders(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// Typecheck without any restrictions.
+	ctx := context.Background()
+	semaCtx := tree.MakeSemaContext()
+	semaCtx.Properties.Require("", 0 /* flags */)
+
+	// Hint all int4 types.
+	placeholderTypes := []*types.T{types.Int4, types.Int4, types.Int4, types.Int4, types.Int4}
+	err := semaCtx.Placeholders.Init(len(placeholderTypes), placeholderTypes)
+	require.NoError(t, err)
+
+	expr, err := parser.ParseExpr("case when 1 < $1 then $2 else $3 end = $4")
+	require.NoError(t, err)
+	typed, err := tree.TypeCheck(ctx, expr, &semaCtx, types.Any)
+	require.NoError(t, err)
+
+	leftTyp := typed.(*tree.ComparisonExpr).Left.(tree.TypedExpr).ResolvedType()
+	require.Equal(t, types.Int4, leftTyp)
+}
+
+func TestTypeCheckCaseExprWithConstantsAndUnresolvedPlaceholders(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// Typecheck without any restrictions.
+	ctx := context.Background()
+	semaCtx := tree.MakeSemaContext()
+	semaCtx.Properties.Require("", 0 /* flags */)
+
+	// Hint all int4 types, but leave one of the THEN branches unhinted.
+	placeholderTypes := []*types.T{types.Int4, types.Int4, types.Int4, nil, types.Int4}
+	err := semaCtx.Placeholders.Init(len(placeholderTypes), placeholderTypes)
+	require.NoError(t, err)
+
+	expr, err := parser.ParseExpr("case when 1 < $1 then $2 when 1 < $3 then $4 else 3 end = $5")
+	require.NoError(t, err)
+	typed, err := tree.TypeCheck(ctx, expr, &semaCtx, types.Any)
+	require.NoError(t, err)
+
+	leftTyp := typed.(*tree.ComparisonExpr).Left.(tree.TypedExpr).ResolvedType()
+	require.Equal(t, types.Int4, leftTyp)
+
+	for i := 0; i < len(placeholderTypes); i++ {
+		pTyp, _, err := semaCtx.Placeholders.Type(tree.PlaceholderIdx(i))
+		require.NoError(t, err)
+		require.Equal(t, types.Int4, pTyp)
+	}
+}

